File: src/Account.cpp
#include "Account.h"
#include <iostream>

Account::Account(int account_id, std::string name,int balance, bool enabled, bool student){
  account_id_ = account_id;
  name_ = name;
  balance_ = balance;
  enabled_ = enabled;
  student_ = student;
}

Account::Account(){}

std::string Account::get_name() {
  return name_;
}

int Account::get_account_id() {
  return account_id_;
}

float Account::get_balance() {
  return balance_;
}

bool Account::is_enabled() {
  return enabled_;
}

bool Account::is_student() {
  return student_;
}

void Account::set_balance(int input) {
  balance_ = input;
}

void Account::enable(bool enable) {
  if (enabled_ == enable && enable == true){
    std::cout << "Transaction denied. Account is already enabled" << std::endl;
  }else if(enabled_ == enable && enable == false){
    std::cout << "Transaction denied. Account is already disabled" << std::endl;
  }else if(enabled_ != enable && enable == true){
    enabled_ = true;
  }else{
    enabled_ = false;
  }
}

void Account::change_plan() {
  if (student_) {
    student_ = false;
  } else {
    student_ = true;
  }
}

File: backend/src/Account.java
package sagyou.jikan;

/*
 * This class holds data for individual accounts
 */
public class Account{
  private int id; // account number
  private String name; // account holder's full name
  private double balance; // account balance
  private boolean enabled; // flag of account if it is enabled
  private boolean student; // flag of account if it has a student plan

  /**
   * Constructor
   * all fields are set at construction of object
   */
  public Account(int id, String name, double balance, boolean enabled, boolean student){
    this.id = id;
    this.name = name;
    this.balance = balance;
    this.enabled = enabled;
    this.student = student;
  }

	/**
	* Returns value of id
	* @return
	*/
	public int getId() {
		return id;
	}

  /**
	* Returns value of name
	* @return
	*/
	public String getName() {
		return name;
	}

  /**
  * Returns value of balance
  * @return
  */
  public double getBalance(){
    return balance;
  }


	/**
	* Returns value of enabled
	* @return
	*/
	public boolean isEnabled() {
		return enabled;
	}

	/**
	* Returns value of student
	* @return
	*/
	public boolean isStudent() {
		return student;
	}

}
